app/models/audit.rb

```diff
@@ -0,0 +1,99 @@
+require 'syslog'
+require 'logger'
+require 'time'
+
+module Audit
+  class << self
+    def notice msg, msgid, data
+      logger.info LogMessage.new msg, msgid, data, Syslog::LOG_NOTICE
+    end
+
+    def logger
+      @logger ||= Rails.logger
+    end
+
+    attr_writer :logger
+  end
+
+  module SDID
+    CONJUR_PEN = 43838
+    def self.conjur_sdid label
+      [label, CONJUR_PEN].join('@').freeze
+    end
+
+    POLICY = conjur_sdid 'policy'
+    AUTH = conjur_sdid 'auth'
+    SUBJECT = conjur_sdid 'subject'
+    ACTION = conjur_sdid 'action'
+  end
+
+  class LogMessage < String
+    def initialize msg, msgid, structured_data = nil, severity = nil
+      super msg
+      @msgid = msgid
+      @structured_data = structured_data
+      @severity = severity
+    end
+
+    attr_reader :msgid, :structured_data, :severity
+  end
+
+  # Middleware to store request ID in a thread variable
+  class RememberUuid < Struct.new :app
+    def call env
+      Thread.current[:request_id] = env["action_dispatch.request_id"]
+      app.call env
+    end
+  end
+
+  class RFC5424Formatter
+    SEVERITY_MAP = {
+      Logger::Severity::DEBUG => Syslog::LOG_DEBUG,
+      Logger::Severity::ERROR => Syslog::LOG_ERR,
+      Logger::Severity::FATAL => Syslog::LOG_CRIT,
+      Logger::Severity::INFO => Syslog::LOG_INFO,
+      Logger::Severity::WARN => Syslog::LOG_WARNING
+    }
+
+    # TODO
+    FACILITY = 32
```

Comment:
Is this actually TODO? Please replace with a comment regarding this constant.

Reply:
Yes, this should be configurable once this code is factored out.